array-manipulation-1/src/compact.js

```diff
@@ -0,0 +1,19 @@
+/* exported compact */
+
+/* define a function named compact with one parameter array
+declare a variable outputArray assigned an array literal
+loop over the array start to finish
+if the current iteration as an index of array is not strictly false or strictly NaN or 0 or -0, or undefined, or ''
+then push the item to outputArray
+return outputArray  */
+
+function compact(array) {
+  var outputArray = [];
+  for (var i = 0; i < array.length; i++) {
+    var x = Boolean(array[i]);
+    if (x) {
+      outputArray.push(array[i]);
+    }
+  }
+  return outputArray;
```

Comment:
You don't actually need to use `Boolean` here! You can just check if each iteration is a truthy or falsy value, since they will have one by default. Example below!
```
function compact(array) {
  var output = [];
  for (var i = 0; i < array.length; i++) {
    var element = array[i];
    if (element) output.push(element);
  }
  return output;
}
```